sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -150,6 +151,16 @@ class CSVOptions(
 
   val isCommentSet = this.comment != '\u0000'
 
+  def dateFormatter: DateTimeFormatter = {
```

Comment:
Why is it `def`?

Reply:
DateTimeFormatter has the disadvantage. It does not implement Serializable in contrast to FastDateFormat. That is why I couldn't make it as a val here.

Reply:
I think you could do this via `lazy val`